static/js/notifications.js

```diff
@@ -99,13 +101,10 @@ export function initialize() {
 
 export function update_notification_sound_source(container_elem, settings_object) {
     const notification_sound = settings_object.notification_sound;
-    const audio_file_without_extension = "/static/audio/notification_sounds/" + notification_sound;
-    container_elem
-        .find(".notification-sound-source-ogg")
-        .attr("src", `${audio_file_without_extension}.ogg`);
-    container_elem
-        .find(".notification-sound-source-mp3")
-        .attr("src", `${audio_file_without_extension}.mp3`);
+    const ogg_file = require("../audio/notification_sounds/" + notification_sound + ".ogg");
+    const mp3_file = require("../audio/notification_sounds/" + notification_sound + ".mp3");
```

Comment:
The modern webpack way would be `await import(…)`, after making the function `async`.

```js
    const {default: ogg_file} = await import(
        `../audio/notification_sounds/${notification_sound}.ogg`
    );
    const {default: mp3_file} = await import(
        `../audio/notification_sounds/${notification_sound}.mp3`
    );
```

(The [Vite way](https://vitejs.dev/guide/assets.html#new-url-url-import-meta-url) is a bit nicer: it recognizes `new URL(…, import.meta.url).href` and compiles it to a string literal.)

Do we actually want to webpackify this though, if we still have to serve the sounds at non-webpack paths as part of the API?

Reply:
I think we still serve the non-webpack paths as part of the API.

Reply:
I don't think we actually care about the hashing -- but we'd like to make the requests to the static domain, if possible, so that we get better cache hit rates from the CDN.

I'm going to drop this commit from the series for now, since I think solving this right gets more complicated.